src/shadowbox/server/server_access_key.ts

```diff
@@ -126,19 +121,30 @@ export class ServerAccessKeyRepository implements AccessKeyRepository {
     }, ServerAccessKeyRepository.QUOTA_ENFORCEMENT_INTERVAL_MS);
   }
 
-  enableSinglePort(portForNewAccessKeys: number) {
-    this.portForNewAccessKeys = portForNewAccessKeys;
+  private isExistingAccessKeyPort(port: number): boolean {
+    return this.accessKeys.some((key) => {
+      return key.proxyParams.portNumber === port;
+    });
+  }
+
+  async setDefaultPort(port: number): Promise<void> {
+    if (!Number.isInteger(port) || port < 1 || port > 65535) {
+      throw new errors.InvalidPortNumber(port);
+    }
+    if (!this.isExistingAccessKeyPort(port) && await isPortUsed(port)) {
+      throw new errors.PortInUse(port);
+    }
+    this.portForNewAccessKeys = port;
```

Comment:
This method should write the config field too, or you may end up in an inconsistent state.
All the other methods persist the changes made.